### DISCUSSION TITLE
Refresh Home Page every 30 seconds

### DISCUSSION THREAD

**Discussion by ozannsk** [Budibase/budibase#5011] [Help]
Hi,

Is there any posibility to refresh home page every 30 seconds or more?

I added refresh button and set action "Refresh Data Provider" into this button. Its okay when clicked on it but is there any posibility to automate this button?

Also, I have a button that navigate to modal in home page. When I changed some values in home page table in this modal, home page table is not refreshed after modal is closed.  Can I refresh automatically the home page after closing the modal?

**Comment by mjashanks** [comment]
Hey @ozannsk - this is not possible today, but we'd like to get this done. It's already logged here: https://github.com/Budibase/budibase/issues/3992

**Comment by eftomi** [comment]
> Also, I have a button that navigate to modal in home page. When I changed some values in home page table in this modal, home page table is not refreshed after modal is closed. Can I refresh automatically the home page after closing the modal?


I have a similar challenge - on /main screen I'm using REST API query data provider to get a list of orders, and a separate modal screen to add an order. When the modal adds an order with another REST API query, it gets closed by Close Screen Modal action and navigates back to /main with Navigate To, however the provider in main screen is not refreshed.

**Reply by AndrewBucklin** [reply]
Also facing the same challenge. Did you find a way to get the table to refresh after the modal screen is closed?
